lib/tilex_web/models/post.ex

```diff
@@ -80,6 +85,27 @@ defmodule Tilex.Post do
     |> hd
   end
 
+  def create_changeset(params, changes) do
+    %__MODULE__{}
+    |> cast(params, permitted_create_params())
+    |> validate_required(required_create_params())
+    |> validate_length(:title, max: title_max_chars())
+    |> validate_length_of_body
+    |> validate_number(:likes, greater_than: 0)
+    |> add_slug
+    |> change(changes)
```

Comment:
@dillonhafer have you considered moving the `changes` and `add_slug` before the validations?

I'd move them between casting and validations. This way you can validate the require on the `developer_id` field as well so you won't rely that you are calling this func with `developer_id` as the changes arg. You'd also not being relying that the `generate_slug` func does not have any bug that could eventually save wrong data into db. There's probably db constraints for that, so we'd be safe in any case, but it's also about providing a good error message to the dev if something goes wrong.

I think that if you'd have a pattern matching on the `changes` arg by for example `[developer_id: developer_id]`, then if you'd call this func on other place of your code and if you forget to add it then the error would be a couldn't find a func that matches your args. You'd see the missing developer_id right away. Other than that you'd see a developer_id is required, so you'd have to investigate who is suppose to be setting this, if it's on cast or change, or somewhere else. This is very minor, I just would like to share my thoughts on this one.

Something like that:

```elixir
def create_changeset(params, developer_id: developer_id) do
  %__MODULE__{}
  |> cast(params, ~w(title body channel_id)a)
  |> change(developer_id: developer_id)
  |> add_slug()
  |> validate_required(~w(title body slug channel_id developer_id)a)
  ...
```

In the end this func is very explicit on what's going on.

Reply:
That pattern matching argument makes really good sense the more I think about it! Thank you Vini!